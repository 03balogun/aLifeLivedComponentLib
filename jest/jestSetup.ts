/* eslint-disable no-undef */
jest.useFakeTimers('legacy'); // Stops reference error from appearing
