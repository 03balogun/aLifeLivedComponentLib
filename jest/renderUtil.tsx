import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';

function render(ui, { ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
