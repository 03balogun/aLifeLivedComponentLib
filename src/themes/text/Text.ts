import { createText } from '@shopify/restyle';
import { Theme } from '../theme';

export const textVariants = {
  h1: {
    fontSize: 32,
    lineHeight: 38,
    color: 'contrastHighest',
  },
  h2: {
    fontSize: 24,
    lineHeight: 30,
    color: 'contrastHighest',
  },
  h3: {
    fontSize: 16,
    lineHeight: 24,
    color: 'contrastHighest',
  },
  body: {
    fontSize: 16,
    lineHeight: 20,
    color: 'contrastHighest',
  },
  caption: {
    fontSize: 14,
    lineHeight: 18,
    color: 'contrastLow',
  },
  default: {
    fontSize: 16,
    lineHeight: 20,
    color: 'contrastHighest',
  },
};

export const Text = createText<Theme>();
