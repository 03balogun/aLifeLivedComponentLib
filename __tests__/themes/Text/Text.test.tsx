import React from 'react';
import { Text } from '../../../src/themes/text/Text';
import { render } from '../../../jest/renderUtil';

describe('', () => {
  describe('h1 Text render', () => {
    test('Render h1 test correctly', () => {
      const { toJSON } = render(<Text variant="h1">H1 variant</Text>);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('h2 Text render', () => {
    test('Render h2 test correctly', () => {
      const { toJSON } = render(<Text variant="h2">H2 variant</Text>);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('h3 Text render', () => {
    test('Render h3 test correctly', () => {
      const { toJSON } = render(<Text variant="h3">H3 variant</Text>);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('Default Text render', () => {
    test('Render default test correctly', () => {
      const { toJSON } = render(<Text variant="default">Default variant</Text>);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('Body Text render', () => {
    test('Render body test correctly', () => {
      const { toJSON } = render(<Text variant="body">Body variant</Text>);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('Caption Text render', () => {
    test('Render caption test correctly', () => {
      const { toJSON } = render(<Text variant="caption">Caption variant</Text>);
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
