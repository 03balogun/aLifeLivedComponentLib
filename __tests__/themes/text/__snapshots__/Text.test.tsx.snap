// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src >> themes >> Text Body Text render Render body test correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#282828",
        "display": "flex",
        "fontFamily": "SFUIDisplay-Regular",
        "fontSize": 18,
      },
    ]
  }
>
  Body variant
</Text>
`;

exports[`src >> themes >> Text BodySmall Text render Render BodySmall test correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#282828",
        "display": "flex",
        "fontFamily": "SFUIDisplay-Regular",
        "fontSize": 16,
      },
    ]
  }
>
  Caption variant
</Text>
`;

exports[`src >> themes >> Text BodyXSmall Text render Render BodyXSmall test correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#282828",
        "display": "flex",
        "fontFamily": "SFUIDisplay-Regular",
        "fontSize": 14,
      },
    ]
  }
>
  Caption variant
</Text>
`;

exports[`src >> themes >> Text Intro Text render Render intro test correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#282828",
        "display": "flex",
        "fontFamily": "SFUIText-Semibold",
        "fontSize": 16,
      },
    ]
  }
>
  Body variant
</Text>
`;

exports[`src >> themes >> Text h1 Text render Render h1 test correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#282828",
        "display": "flex",
        "fontFamily": "SFUIText-Bold",
        "fontSize": 52,
      },
    ]
  }
>
  H1 variant
</Text>
`;

exports[`src >> themes >> Text h2 Text render Render h2 test correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#282828",
        "display": "flex",
        "fontFamily": "SFUIText-Bold",
        "fontSize": 32,
      },
    ]
  }
>
  H2 variant
</Text>
`;

exports[`src >> themes >> Text h3 Text render Render h3 test correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#282828",
        "display": "flex",
        "fontFamily": "SFUIText-Bold",
        "fontSize": 24,
      },
    ]
  }
>
  H3 variant
</Text>
`;

exports[`src >> themes >> Text h4 Text render Render h4 test correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#282828",
        "display": "flex",
        "fontFamily": "SFUIText-Regular",
        "fontSize": 20,
      },
    ]
  }
>
  H3 variant
</Text>
`;
